subiquity/server/controllers/install.py

```diff
@@ -668,6 +668,24 @@ async def platform_postinstall(self):
                     log.warning("chreipl stderr:\n%s", cpe.stderr)
                 raise
 
+    def kernel_components(self) -> List[str]:
+        if not self.supports_apt():
+            return []
+        if not self.model.drivers.do_install:
+            return []
+        info: VariationInfo = self.app.controllers.Filesystem._info
+        kernel_components = info.available_kernel_components
+        for driver in self.app.controllers.Drivers.drivers:
+            m = re.match("nvidia-driver-([0-9]+)", driver)
+            if not m:
+                continue
+            v = m.group(1)
+            ko = f"nvidia-{v}-ko"
+            user = f"nvidia-{v}-user"
+            if ko in kernel_components and user in kernel_components:
+                return [ko, user]
```

Comment:
thought: I think the driver list is generally ascending so that means we'll get the oldest compatible driver first here. I don't know how many kernel component versions we'll have so maybe it doesn't actually matter in practice.

Reply:
I suspect that only one component version is ever going to be offered, I'm checking on that.
I could do a reverse sort here and cause us to prefer the larger number over the smaller one, which sounds like good defensive programming anyhow.  Will implement.